tfhe/src/integer/server_key/radix_parallel/count_zeros_ones.rs

```diff
@@ -0,0 +1,543 @@
+                || {
+                    let lut = self.key.generate_lookup_table(|x| {
+                        // extract message
+                        let x = x % self.key.message_modulus.0 as u64;
+                        // bitnot the message
+                        (!x) % self.key.message_modulus.0 as u64
+                    });
+                    result
+                        .blocks
+                        .par_iter()
+                        .map(|block| self.key.apply_lookup_table(block, &lut))
+                        .collect::<Vec<_>>()
+                },
+                || {
+                    let lut = self.key.generate_lookup_table(|x| {
+                        // extract carry
+                        let x = x / self.key.message_modulus.0 as u64;
+                        // bitnot the carry
+                        (!x) % self.key.message_modulus.0 as u64
+                    });
+                    let mut carry_blocks = Vec::with_capacity(num_unsigned_blocks);
+                    result.blocks[..num_signed_blocks - 1] // last carry is not interesting
+                        .par_iter()
+                        .map(|block| self.key.apply_lookup_table(block, &lut))
+                        .collect_into_vec(&mut carry_blocks);
+                    // Normally this would be 0, but we want the bitnot of 0, which is msg_mod-1
+                    carry_blocks.insert(
+                        0,
+                        self.key
+                            .create_trivial((self.message_modulus().0 - 1) as u64),
+                    );
+                    carry_blocks
+                },
+            );
+            let message = SignedRadixCiphertext::from(message_blocks);
+            let carry = SignedRadixCiphertext::from(carry_blocks);
+            let result = self
+                .sum_ciphertexts_parallelized(
+                    [
+                        message,
+                        carry,
+                        self.create_trivial_radix(2u32, num_signed_blocks),
+                    ]
+                    .iter(),
+                )
+                // Go back to unsigned world because we know the value cannot be negative
+                // but casting from signed to unsigned may require to look at the sign bit
+                // which we know is not set
+                .map(|ct| RadixCiphertext::from(ct.blocks))
+                .unwrap();
+
+            self.cast_to_unsigned(result, min_num_blocks_to_have_32_bits as usize)
+        }
+    }
+
+    /// More complex implementation of count zeros/ones meant for 2_2 parameters
+    ///
+    /// * It will only work for 2_2 parameters
+    /// * ct must not have any carries
+    /// * The returned result has enough blocks to encrypt 32bits (e.g. 1_1 parameters -> 32 blocks,
+    ///   3_3 parameters -> 11 blocks == 33 bits)
+    fn count_bits_2_2<T>(&self, ct: &T, count_kind: BitCountKind) -> RadixCiphertext
```

Comment:
perf gains of this variant ?

Reply:
As said in the commit its  none on small sizes (< 64bits) and rather small on 64 bits (iirc naive 64 bits its ~135 ms, while non naive is ~115ms), where its really interesting is for really big sizes like 12800 bits where we go from 1.8s for naive, 1.1 sec for non-naive likely because the number of pbs reduced starts to be noticeable and there are less things to sum